src/librustc/middle/cstore.rs

```diff
@@ -474,7 +477,13 @@ pub mod tls {
     {
         let tls_payload = (ecx as *const _, encoder as *mut _);
         let tls_ptr = &tls_payload as *const _ as *const TlsPayload;
-        TLS_ENCODING.set(unsafe { &*tls_ptr }, || f(ecx, encoder))
+        TLS_ENCODING.with(|tls| {
+            let prev = tls.get();
```

Comment:
This is repeated at least three times in the PR. Any lightweight abstraction to be had (short of full scoped TLS)? I believe @nikomatsakis actually wanted to continue using scoped TLS but was OK ultimately using an external lib. Since we can't do that yet, it'd be nice to have a cleaner solution in the meantime.